src/app/template-library/template-library.service.ts

```diff
@@ -0,0 +1,144 @@
+import {Injectable} from '@angular/core';
+import {map} from 'rxjs/operators';
+import {environment} from '@environments/environment';
+import {DemoService} from '@services/demo.service';
+import {RequestService} from '../shared/request/request.service';
+import {Observable} from 'rxjs/Observable';
+
+export interface Template {
+  uuid: string;
+  name: string;
+  description?: string;
+  leadImageUrl?: string;
+  leadVideoUrl?: string;
+  institutionUUID: string;
+  isPublic?: boolean;
+  type?: string;
+  attributes: string[];
+  modified?: string;
+}
+
+export interface Category {
+  name: string;
+  type: string;
+  leadImage: string;
+  color: string;
+  size: Size;
+}
+
+export interface CategorisedTemplates {
+  category: Category;
+  templates: Template[];
+}
+
+enum Size {
+  'LARGE', 'SMALL'
+}
+
+@Injectable({
+  providedIn: 'root'
+})
+export class TemplateLibraryService {
+
+  constructor(
+    private request: RequestService,
+    private demo: DemoService,
+  ) { }
+
+  getTemplates(): Observable<Template[]> {
+    if (environment.demo) {
+      return this.demo.getTemplates().pipe(map(this._handleTemplates));
+    }
+    return this.request.graphQLQuery(
+      `query templates {
+        templates {
+          uuid
+          name
+          description
+          leadImageUrl
+          leadVideoUrl
+          institutionUUID
+          isPublic
+          type
+          attributes {
+            name
+          }
+          modified
+        }
+      }`,
+      {},
+      {
+        noCache: true,
+      }
+    ).pipe(map(this._handleTemplates));
+  }
+
+  private _handleTemplates(res) {
+    if (!res.data) {
+      return [];
+    }
+    return res.data.templates.map(template => {
+      return {
+        ...template,
+        ...{
+          attributes: template.attributes.map(t => t.name)
+        }
+      };
+    });
+  }
+
+  getCategories(): Category[] {
+    return [
+      {
+        'leadImage': '',
+        'name': 'Team Projects',
+        'type': 'team project',
+        'color': 'rgba(0,64,229, 0.7)',
+        'size': Size.LARGE
```

Comment:
It's not clear that `0` means large and `1` means small.
We can simply change it to `isLarge: boolean` to make it more clear